docs/consensus/consensus_chain.md

```diff
@@ -109,12 +109,10 @@ Justifications contain a set of all PoT checkpoints since the parent block up to
 ## Fast sync
 
 1. Obtain segment headers from DSN: 1-4 from [sync from DSN implementation](#Synchronization)
-2. If the last segment index is greater than 2 - download the last two segments from DSN otherwise exit [Fast sync](#fast-sync) and continue with either [sync from DSN implementation](#Synchronization) or [Substrate Sync](#substrate-sync) .
-3. Reconstruct blocks from the two downloaded segments.
-4. Import the last block from the second last segment into the blockchain DB bypassing the blockchain checks of missing parent block.
-5. Import the state using [Substrate Sync](#substrate-sync)  for the newly imported block.
-6. Import and execute all blocks from the last segment in regular mode.
-7. Pass the control to [sync from DSN implementation](#Synchronization). It will either download the new archived segment if any or pass the control to [Substrate Sync](#substrate-sync).
+2. If the last segment index is greater than 2 - download the last two segments from DSN otherwise exit [Fast sync](#fast-sync) and continue with either [sync from DSN implementation](#Synchronization) or [Substrate Sync](#substrate-sync) . Reconstruct blocks from the two downloaded segments. Note: we need only a single last block from the second last segment and we can optimize the solution if acquire this block from Substrate network instead.
+3. Import the last block from the second last segment into the blockchain DB bypassing the blockchain checks of missing parent block. Import the state using [Substrate Sync](#substrate-sync) for the newly imported block. Note: block and its state should be imported as atomic operation.
+4. Import and execute all blocks from the last segment in regular mode.
+5. Pass the control to [sync from DSN implementation](#Synchronization). It will either download the new archived segment if any or pass the control to [Substrate Sync](#substrate-sync).
```

Comment:
This repeats previous item somewhat and can be made more conscise, also when downloading state we are talking about state sync specifically, not what "Substrate Sync" section below describes:
```suggestion
3. Download state that corresponds to the first block received in the previous step using Substrate State Sync
4. Import the first block of the last segment with its state into the blockchain DB bypassing the blockchain checks of missing parent block, it is important for this to be an atomic operation
5. Import and execute other remaining blocks from the last segment as they would normally
6. Pass the control to [sync from DSN implementation](#Synchronization). It will either download the new archived segment if any or pass the control to [Substrate Sync](#substrate-sync).
```